### PR TITLE
[Feature] Support the basic dbt 1.6 metric

### PR DESCRIPTION
**Pull request by wcchang1115**
**PR checklist**

- [x] Ensure you have added or ran the appropriate tests for your PR.
- [x] DCO signed

**What type of PR is this?**
feature

**What this PR does / why we need it**:
- Because dbt 1.6 introduced the new dbt metric, it's the update to support new metric query.

**Which issue(s) this PR fixes**:
sc-32064

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

NONE

### CODE REVIEW

File: piperider_cli/__init__.py [hunk 1]
```diff
@@ -216,6 +219,7 @@ def silent_none(var):
         return var
 
     env.globals['env_var'] = env_var
+    env.globals['Dimension'] = dimension
```
Comment (popcornylu) [inline piperider_cli/__init__.py:222]: Is it better not to share the template render logic from dbt config and metric filter?

File: piperider_cli/dbtutil.py [hunk 2]
```diff
@@ -350,6 +351,245 @@ def is_chosen(key, metric):
...
+                filter=denominator.get('filter'),
+                alias=denominator.get('alias'))
+            ref_metrics.append(m2)
+            derived_time_grains = find_derived_time_grains(manifest, metric_map[denominator.get('name')])
+            if len(time_grains) < len(derived_time_grains):
+                time_grains = derived_time_grains
+
+            m = Metric(metric.get('name'),
+                       calculation_method='derived',
+                       expression=f"{numerator.get('name')} / {denominator.get('name')}",
```
Comment (popcornylu) [inline piperider_cli/dbtutil.py:567]: nitpick: Would be better to pass the calculation_method & numerator & denominator to Metric object. Then generate the SQL accordingly. But now, it is fine.

File: piperider_cli/__init__.py (hunk 1 above)
Comment (wcchang1115) [reply piperider_cli/__init__.py:222]: yes, agree with you that it's better to separate

File: piperider_cli/dbtutil.py (hunk 2 above)
Comment (wcchang1115) [reply piperider_cli/dbtutil.py:580]: Will put it into another story to enhance

**Pull request merged**